lumos/optimal_control/scaled_mesh_ocp.py

```diff
@@ -633,19 +625,74 @@ def _set_scales(self, scales: Dict[str, Dict[str, float]]):
         to minimize the changes required to NLP functions.
         """
 
-        self.model.set_scales(scales)
-        # Decision variable scale from stage var
+        op = self.dec_var_operator
+
+        # First we create a dictioinary to summarise all scales in one place.
         var_scales = {
-            g: np.tile(1 / self.model.get_group_scales(g), (self.num_stages, 1))
-            for g in self.stage_var_groups
+            g: self.model.make_const_vector(g, 1.0) for g in self.model._implicit_inputs
         }
-        # TODO: We should set this somewhere else
         for var_name in self.global_var_names:
-            if var_name in scales:
-                var_scales[var_name] = scales[var_name]
-            else:
-                var_scales[var_name] = 1
-        self._dec_var_scales = self.dec_var_operator.flatten_var(**var_scales)
+            var_scales[var_name] = 1.0
+
+        # create default decision variable scales
+        self._dec_var_scales = np.ones(self.num_dec)
+
+        # Overwrite with scales from configs and update decision variable scales
+        for sc in scale_configs:
+            if isinstance(sc, GlobalVarScaleConfig):
+                var_scales[sc.name] = sc.value
+                self._dec_var_scales[op.get_global_var_index(sc.name)] = sc.value
+            elif isinstance(sc, StageVarScaleConfig):
+                var_scales[sc.group][
+                    op.get_var_index_in_group(sc.group, sc.name)
+                ] = sc.value
+                self._dec_var_scales[
+                    op.get_var_index_in_dec(sc.group, sc.name)
+                ] = sc.value
+
+        # Set constarint scales
+
+        continuity_scales = np.ravel(
+            np.tile(
+                var_scales["states"],
+                (self.num_intervals, op.num_stages_per_interval - 1, 1),
+            )
+        )
+        if self.is_condensed:
```

Comment:
add some comments about how the condensed constraints should be scaled (the stored constraints are unscaled, etc...)

Reply:
done